spec/rest.spec.js

```diff
@@ -623,5 +623,48 @@ describe('rest update', () => {
       done();
     });
   });
+});
+
+describe('read-only masterKey', () => {
+  it('properly throws on create using readOnly masterKey', () => {
+    const config = Config.get('test');
+    const readOnly = auth.readOnly(config);
+    expect(() => {
+      rest.create(config, readOnly, 'AnObject', {})
+    }).toThrow(new Parse.Error(Parse.Error.OPERATION_FORBIDDEN, `read-only masterKey isn't allowed to perform the create operation.`));
+    expect(() => {
+      rest.update(config, readOnly, 'AnObject', {})
+    }).toThrow();
+    expect(() => {
+      rest.del(config, readOnly, 'AnObject', {})
+    }).toThrow();
+  });
 
```

Comment:
Looks like we still need some more tests in here to verify that read still works, config updates are blocked & direct schema updates are blocked

Reply:
yes I'm on it!